contracts/SponsorFaucet.sol

```diff
@@ -35,27 +35,21 @@ contract SponsorFaucet is Ownable, Pausable, ReentrancyGuard {
     }
      
     /*** Dapp dev calls ***/ 
-    //apply cfx for gas & storage
-    function applyFor(address dapp) public nonReentrant whenNotPaused {
-        cpc.setSponsorForGas.value(gas_bound)(dapp, upper_bound);
-        cpc.setSponsorForCollateral.value(collateral_bound)(dapp);
-        dapps[dapp].gas_cnt.add(1);
-        dapps[dapp].collateral_cnt.add(1);
-        dapps[dapp].limit.add(gas_bound.add(collateral_bound));
-        emit applied(msg.sender, dapp, upper_bound);
-    }
-
     function applyForGas(address dapp) public nonReentrant whenNotPaused {
+        require(address(this).balance >= gas_bound, "faucet out of money");
```

Comment:
why do you check for the faucet contract balance against gas_bound?

Reply:
1. faucet contract should have enough balance for users
2. gas_bound is the amount of funds provided for each application